strawberry/django/views.py

```diff
@@ -99,7 +100,12 @@ def _create_response(
         self, response_data: GraphQLHTTPResponse, sub_response: HttpResponse
     ) -> JsonResponse:
         response = JsonResponse(response_data)
-        response._headers.update(sub_response._headers)
+
+        if django.VERSION >= (3, 2):
+            for name, value in sub_response.headers.items():
+                response.headers[name] = value
```

Comment:
```suggestion
            for name, value in sub_response.headers.items():
                response[name] = value
```

We could always do this right? Regardless of the Django version.

Reply:
Indeed!

See:
https://github.com/django/django/blob/3.1.7/django/http/response.py#L135
https://github.com/django/django/blob/3.2rc1/django/http/response.py#L72
https://github.com/django/django/blob/3.2rc1/django/http/response.py#L171

Reply:
Sorry this won't work actually because `sub_response.headers` is not available pre 3.2. 

It also looks like the `.items` property (which is what I would expect to be able to use) has actually changed in 3.2 which I think is a breaking change. Not sure if thats ok for Django to do that...
https://github.com/django/django/blob/3.1.7/django/http/response.py#L152-L153
https://github.com/django/django/blob/3.2rc1/django/http/response.py#L186-L187

Reply:
Anyway you're original suggestion is probably the best way of doing this now @AGASS007 . Sorry for the churn!

Reply:
> Sorry this won't work actually because `sub_response.headers` is not available pre 3.2.
> 
> It also looks like the `.items` property (which is what I would expect to be able to use) has actually changed in 3.2 which I think is a breaking change. Not sure if thats ok for Django to do that...
> [https://github.com/django/django/blob/3.1.7/django/http/response.py#L152-L153](https://github.com/django/django/blob/3.1.7/django/http/response.py?rgh-link-date=2021-03-19T11%3A47%3A18Z#L152-L153)
> [https://github.com/django/django/blob/3.2rc1/django/http/response.py#L186-L187](https://github.com/django/django/blob/3.2rc1/django/http/response.py?rgh-link-date=2021-03-19T11%3A47%3A18Z#L186-L187)

We could ask about this on the developer mailing list

Reply:
Grrrrr `items()` on `HttpResponse` returns headers `items()`, but used to return `values()`

https://github.com/django/django/blob/3.1.7/django/http/response.py#L152
https://github.com/django/django/blob/3.2rc1/django/http/response.py#L186

Reply:
Nevermind, for versions < 3.2, `values()` returns an iterable of `(key, value)` so the output of `HttpResponse.items()` is consistant across versions.